File: erp_flutter/lib/core/models/document.dart
class Document {
  final int id;
  final String title;
  final String category;
  final String fileType;
  final int fileSize;
  final String uploadedBy;
  final DateTime uploadDate;
  final int version;
  final String status;
  final String department;
  final DateTime createdAt;

  Document({
    required this.id,
    required this.title,
    required this.category,
    required this.fileType,
    required this.fileSize,
    required this.uploadedBy,
    required this.uploadDate,
    required this.version,
    required this.status,
    required this.department,
    required this.createdAt,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      category: json['category'] ?? '',
      fileType: json['fileType'] ?? '',
      fileSize: json['fileSize'] ?? 0,
      uploadedBy: json['uploadedBy'] ?? '',
      uploadDate: json['uploadDate'] != null ? DateTime.parse(json['uploadDate']) : DateTime.now(),
      version: json['version'] ?? 1,
      status: json['status'] ?? '',
      department: json['department'] ?? '',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : DateTime.now(),
    );
  }
}

File: erp_api/ERP.API/Features/Documents/Document.cs
namespace ERP.API.Features.Documents;

public class Document
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty; // Policy, Contract, Report, Memo, Letter, Manual
    public string FileType { get; set; } = string.Empty; // PDF, DOCX, XLSX, PPTX
    public long FileSize { get; set; } // in bytes
    public string UploadedBy { get; set; } = string.Empty;
    public DateTime UploadDate { get; set; }
    public int Version { get; set; } = 1;
    public string Status { get; set; } = "Active"; // Active, Archived, PendingReview, Draft
    public string Department { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
